src/app/components/category-products/category-products.component.ts

```diff
@@ -27,8 +46,104 @@ export class CategoryProductsComponent {
     switchMap((id) => this._categoryService.getOnecategory(id))
   );
 
+  readonly sort$: Observable<string> = this.sort.valueChanges.pipe(
+    startWith(this.sort.value)
+  );
+
+  readonly categoryProducts$: Observable<ProductModel[]> = combineLatest([
+    this.categoryId$,
+    this._productService.getAll(),
+    this.sort$,
+  ]).pipe(
+    map(([catId, products, sort]) => {
+      const prods = products.filter((product) => product.categoryId === catId);
+      if (sort === 'Featured')
```

Comment:
you can simplify this code if you add more info to your sort query model. Technically you can only do it with a single if statement